samples/unit-tests/indicator-dmi/recalculations/demo.js

```diff
@@ -0,0 +1,266 @@
+QUnit.test(
+    'Testing DMI indicator (values, period and updates), #15140.',
+    assert => {
+        const chart = Highcharts.stockChart('container', {
+                yAxis: [{
+                    height: '60%'
+                }, {
+                    height: '40%',
+                    top: '60%'
+                }],
+                series: [{
+                    id: 'main',
+                    type: 'ohlc',
+                    data: [
+                        [1552311000000, 43.87, 44.78, 43.84, 44.72],
+                        [1552397400000, 45.00, 45.67, 44.84, 45.23],
+                        [1552483800000, 45.56, 45.83, 45.23, 45.43],
+                        [1552570200000, 45.97, 46.03, 45.64, 45.93],
+                        [1552656600000, 46.21, 46.83, 45.94, 46.53],
+                        [1552915800000, 46.45, 47.1, 46.45, 47.01],
+                        [1553002200000, 47.09, 47.25, 46.48, 46.63],
+                        [1553088600000, 46.56, 47.37, 46.18, 47.04],
+                        [1553175000000, 47.51, 49.08, 47.45, 48.77],
+                        [1553261400000, 48.83, 49.42, 47.69, 47.76],
+                        [1553520600000, 47.88, 47.99, 46.65, 47.19],
+                        [1553607000000, 47.92, 48.22, 46.15, 46.7],
+                        [1553693400000, 47.19, 47.44, 46.64, 47.12],
+                        [1553779800000, 47.24, 47.39, 46.88, 47.18],
+                        [1553866200000, 47.46, 47.52, 47.13, 47.49],
+                        [1554125400000, 47.91, 47.92, 47.1, 47.81],
+                        [1554211800000, 47.77, 48.62, 47.76, 48.51],
+                        [1554298200000, 48.31, 49.13, 48.29, 48.84],
+                        [1554384600000, 48.7, 49.09, 48.28, 48.92],
+                        [1554471000000, 49.11, 49.28, 48.98, 49.25],
+                        [1554730200000, 49.1, 50.06, 49.08, 50.03],
+                        [1554816600000, 50.08, 50.71, 49.81, 49.88],
+                        [1554903000000, 49.67, 50.19, 49.54, 50.15],
+                        [1554989400000, 50.21, 50.25, 49.61, 49.74],
+                        [1555075800000, 49.8, 50.03, 49.05, 49.72]
+                    ]
+                }, {
+                    type: 'dmi',
+                    linkedTo: 'main',
+                    yAxis: 1
+                }]
+            }),
+            DMIIndicator = chart.series[1],
+            testNumberOfPoints = textDescription => {
+                assert.strictEqual(
+                    chart.series[0].points.length,
+                    DMIIndicator.points.length +
+                        DMIIndicator.options.params.period,
+                    textDescription
+                );
+            };
+
+        testNumberOfPoints(
```

Comment:
An interesting idea for writing the tests! 
I've never seen this before ;)
However, I would choose one way and remain constant through the file.

Reply:
I'm gonna defend this. We have plenty of tests where, for example, asserts are inside `forEach` instead of repeating them one by one. I know this case is slightly different, but it's not wrong - I test the same assert 6 times so using a function to make it a little bit shorter seems fine to me.

However, if others share your opinion, I will change it.
@pawellysy @pawelfus what do you think?

Reply:
The two concerns I have about it are:
- Arrow functions are not supported by Internet Explorer. So it will throw a syntax error.
- Inconsistent syntax could make it harder to refactor all tests to a different setup in the future.

Reply:
These are only tests, not official demos - more and more tests are written in ES6, so the IE should not be a concern here.
When it comes to future refactoring - good point, didn't know about refactoring plans. If there are any, you could make a short note and let us know how you would like us to write tests and what should we pay attention to.